### PR TITLE
Added validation to the promo code field

### PR DESCRIPTION
**Pull request by PrudinnikDmitriy**
https://app.clickup.com/t/3206k3q

### CODE REVIEW

**Comment by PrudinnikDmitriy** [comment]
https://app.clickup.com/t/3206k3q

**Review by AndriiHrohodza — CHANGES_REQUESTED** [review]

File: assets/black-friday-scripts.js [hunk 1]
```diff
@@ -37,11 +38,19 @@ $(document).ready(function() {
     var value = $(this).data('value');
     const collectionHref = $(this).data('collectionHref')
 
-    if (discount != '') {
-      document.cookie = 'discount_code=' + discount + '; expires=0; path=/';
+    document.getElementById('discountInput').onkeyup = function(event) {
+      if (this.value.length === 0) {
```
[the remark this answers is not in the document]
Comment (PrudinnikDmitriy) [reply assets/black-friday-scripts.js]: @AndriiHrohodza  fixed

**Review by AndriiHrohodza — APPROVED** [review]

**Comment by yevheniitsuryk** [comment]
@PrudinnikDmitriy https://bystrov.slack.com/archives/C03RHD1CESC/p1667221473783299

File: assets/black-friday-scripts.js [hunk 2]
```diff
@@ -33,15 +33,21 @@ $(document).ready(function() {
...
+    var discount = $(this).siblings('input[name="discount_code"]').val();
     var value = $(this).data('value');
     const collectionHref = $(this).data('collectionHref')
 
-    if (discount != '') {
-      document.cookie = 'discount_code=' + discount + '; expires=0; path=/';
-
+    document.getElementById('discountInput').onkeyup = function(event) {
+      if ( !this.value.length ) {
+        document.getElementById('textOfError').style.visibility = "hidden";
```
Comment (yevheniitsuryk) [inline assets/black-friday-scripts.js:42]: @PrudinnikDmitriy formatting please
```suggestion
        document.getElementById('textOfError').style.visibility = 'hidden';
```

File: assets/black-friday-scripts.js [hunk 3]
```diff
@@ -33,15 +33,21 @@ $(document).ready(function() {
...
+      }
+    }
+    if (discount.startsWith('SLOW')) {
+      document.cookie = 'discount_code=' + discount + '; expires=0; path=';
       window.location.href = collectionHref;
-    } 
+    } else{
```
Comment (yevheniitsuryk) [inline assets/black-friday-scripts.js:48]: @PrudinnikDmitriy here too
```suggestion
    } else {
```

**Review by yevheniitsuryk — CHANGES_REQUESTED** [review]

File: assets/black-friday-scripts.js (hunk 2 above)
Comment (PrudinnikDmitriy) [reply assets/black-friday-scripts.js]: @yevheniitsuryk  fixed

File: assets/black-friday-scripts.js (hunk 3 above)
Comment (PrudinnikDmitriy) [reply assets/black-friday-scripts.js]: @yevheniitsuryk fixed

**Review by yevheniitsuryk — APPROVED** [review]

**Pull request merged**